bindings/python/pinocchio/visualize/base_visualizer.py

```diff
@@ -54,48 +70,111 @@ def loadViewerModel(self, *args, **kwargs):
         """Create the scene displaying the robot meshes in the viewer"""
         pass
 
-    def reload(self, new_geometry_object, geometry_type = None):
-        """ Reload a geometry_object given by its type"""
+    def reload(self, new_geometry_object, geometry_type=None):
+        """Reload a geometry_object given by its type"""
         pass
 
     def clean(self):
-        """ Delete all the objects from the whole scene """
+        """Delete all the objects from the whole scene"""
         pass
 
-    def display(self, q = None):
+    def display(self, q=None):
         """Display the robot at configuration q or refresh the rendering
         from the current placements contained in data by placing all the bodies in the viewer."""
         pass
 
-    def displayCollisions(self,visibility):
+    def displayCollisions(self, visibility):
         """Set whether to display collision objects or not."""
         pass
- 
-    def displayVisuals(self,visibility):
+
+    def displayVisuals(self, visibility):
         """Set whether to display visual objects or not."""
-        pass
+        raise NotImplementedError()
+
+    def setBackgroundColor(self, *args, **kwargs):
+        """Set the visualizer background color."""
+        raise NotImplementedError()
+
+    def setCameraTarget(self, target):
+        """Set the camera target."""
+        raise NotImplementedError()
 
-    def captureImage(self):
+    def setCameraPosition(self, position: np.ndarray):
+        """Set the camera's 3D position."""
+        raise NotImplementedError()
+
+    def setCameraZoom(self, zoom: float):
+        """Set camera zoom value."""
+        raise NotImplementedError()
+
+    def setCameraPose(self, pose: np.ndarray = np.eye(4)):
+        """Set camera 6D pose using a 4x4 matrix."""
+        raise NotImplementedError()
+
+    def captureImage(self, w=None, h=None):
         """Captures an image from the viewer and returns an RGB array."""
         pass
 
+    def disableCameraControl(self):
+        raise NotImplementedError()
+
+    def enableCameraControl(self):
+        raise NotImplementedError()
+
+    def drawFrameVelocities(self, *args, **kwargs):
+        """Draw current frame velocities."""
+        raise NotImplementedError()
+
     def sleep(self, dt):
         time.sleep(dt)
 
-    def play(self, q_trajectory, dt, capture=False):
+    def play(self, qs, dt=None, callback=None, **kwargs):
```

Comment:
This would break existing code with `capture=True`, right ?

Reply:
Oh yeah, that would be a breaking change, since now whether or not we capture footage is controlled by the `VideoContext` context manager.

What we can do is check for "capture" inside `kwargs` and in that case raise a warning, maybe ?

Reply:
I would think warnings are for deprecation, here this should be an error, because the breaking change would already be activated.

To avoid breaking changes, we should also return the list of images, and not require imageio, which is a third-party library (which we would have to add on all package managers). Also imageio is not compatible with python 2.

Reply:
We could make imageio optional, no problem.

Maybe make getting the images optional (for long streams of images it would be more efficient to just append them to the imageio stream).

My suggestion would be: use the new VideoContext by default when available, otherwise return the images.